packages/sync-actions/src/utils/common-actions.js

```diff
@@ -11,17 +17,30 @@ import * as diffpatcher from './diffpatcher'
  * @param  {Object} options.diff - the diff object
  * @param  {Object} options.oldObj - the object that needs to be updated
  * @param  {Object} options.newObj - the new representation of the object
+ * @param {Boolean} options.shouldOmitEmptyString - a flag to determine if we should treat an empty string a NON-value
  */
-export function buildBaseAttributesActions({ actions, diff, oldObj, newObj }) {
+export function buildBaseAttributesActions({
+  actions,
+  diff,
+  oldObj,
+  newObj,
+  shouldOmitEmptyString,
+}) {
   return actions
     .map(item => {
       const key = item.key // e.g.: name, description, ...
       const actionKey = item.actionKey || item.key
       const delta = diff[key]
       const before = oldObj[key]
       const now = newObj[key]
-      const isNotDefinedBefore = isNil(oldObj[key])
-      const isNotDefinedNow = isNil(newObj[key])
+      const isNotDefinedBefore = getIsAnEmptyValue(
+        oldObj[key],
+        shouldOmitEmptyString
```

Comment:
With my comment above this could become: `shouldOmitEmptyString ? [undefined, null, ''] : [undefined, null]`.